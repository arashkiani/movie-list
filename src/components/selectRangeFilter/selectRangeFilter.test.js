/*global shallow, mount, sinon, store, mockStore */
import {SelectRangeFilter} from './index'

const mapToProps = {
	filterByRange:  3,
	updateRangeFilter: jest.fn(({target}) => {
		mapToProps.filterByRange = parseInt(target.value, 10)
	})
}
let shallowWrapper;
beforeEach(() => {
	shallowWrapper = shallow(<SelectRangeFilter {...store} {...mapToProps}/>)
})
describe('<SelectRangeFilter />', () => {
	it('Match Snapshot', () => {
		expect(shallowWrapper).toMatchSnapshot()
	})
	it('change range from 3 to 5', () => {
		const selectedInput = shallowWrapper.find('form').first()
		selectedInput.simulate('change', { target: { value: 5 } })
		expect(mapToProps.filterByRange).toBe(5)
	})
})