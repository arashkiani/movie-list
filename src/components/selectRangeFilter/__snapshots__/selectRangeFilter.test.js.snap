// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectRangeFilter /> Match Snapshot 1`] = `
<ForwardRef>
  <form
    onChange={[Function]}
  >
    <ForwardRef
      htmlFor="minRating"
    >
      Rating
    </ForwardRef>
    <Component
      defaultValue={3}
      max="10"
      min="0"
      name="minRating"
      step="0.5"
      type="range"
    />
  </form>
</ForwardRef>
`;
